tests/test_pipelines_automatic_speech_recognition.py

```diff
@@ -15,20 +15,56 @@
 import unittest
 
 from transformers import AutoFeatureExtractor, AutoTokenizer, Speech2TextForConditionalGeneration, Wav2Vec2ForCTC
-from transformers.pipelines import AutomaticSpeechRecognitionPipeline
+from transformers.pipelines import AutomaticSpeechRecognitionPipeline, pipeline
 from transformers.testing_utils import require_datasets, require_torch, require_torchaudio, slow
 
 
+# We can't use this mixin because it assumes TF support.
 # from .test_pipelines_common import CustomInputPipelineCommonMixin
 
 
 class AutomaticSpeechRecognitionPipelineTests(unittest.TestCase):
-    # pipeline_task = "automatic-speech-recognition"
-    # small_models = ["facebook/s2t-small-mustc-en-fr-st"]  # Models tested without the @slow decorator
-    # large_models = [
-    #     "facebook/wav2vec2-base-960h",
-    #     "facebook/s2t-small-mustc-en-fr-st",
-    # ]  # Models tested with the @slow decorator
+    @require_torch
+    @slow
+    def test_pt_defaults(self):
+        pipeline("automatic-speech-recognition", framework="pt")
+
+    @require_torch
+    def test_torch_small(self):
+        import numpy as np
+
+        nlp = pipeline(
+            task="automatic-speech-recognition",
+            model="facebook/s2t-small-mustc-en-fr-st",
+            tokenizer="facebook/s2t-small-mustc-en-fr-st",
+            framework="pt",
+        )
+        waveform = np.zeros((34000,))
+        output = nlp(waveform)
+        self.assertEqual(output, {"text": "C'est ce que j'ai fait à ce moment-là."})
+
+    @require_datasets
+    @require_torch
+    @slow
+    def test_torch_large(self):
+        import numpy as np
+
+        nlp = pipeline(
```

Comment:
```suggestion
        speech_recognizer = pipeline(
```

Reply:
We all greed to stop naming the pipelines `nlp` so I really insist on naming this one differently :-)
I know there are others in the tests and docs, that was not a very successful first issue. Will fix when I have a bit of time.

Reply:
Thanks, this escaped the rework.